app/assets/javascripts/views/time_table_by_week.js

```diff
@@ -0,0 +1,187 @@
+        $td = $('<td>').appendTo($tr);
+        $td.attr('colspan', this.colSpan*cellCounter);
+        $td.text(month);
+        month = date.format(monthFormat);
+        cellCounter = 0;
+      }
+      cellCounter++;
+    }
+
+    if (cellCounter > 0) {
+        $td = $('<td>').appendTo($tr);
+        $td.attr('colspan', this.colSpan*cellCounter);
+        $td.text(month);
+    }
+  },
+
+  drawWeeks: function() {
+    var $tr = null,
+        $td = null,
+        cols = this.getNumberOfWeeks(),
+        beginDate = this.timeTableDateRange.begin;
+
+    $tr = $('<tr>').appendTo(this.$table);
+    $tr.addClass('head');
+
+    for (var col = 0; col < cols; col++) {
+      $td = $('<td>').appendTo($tr).attr('colspan', this.colSpan);
+      $td.text(moment(beginDate).add(col,'weeks').format('W'));
+    }
+  },
+
+  drawEmptyTable: function() {
+    var $tr = null,
+        $td = null,
+        date = null,
+        cols = this.getNumberOfWeeks(),
+        weekStart = moment(this.timeTableDateRange.begin).startOf('isoweek'),
+        cellID = '';
+
+    this.members.each( function(user) {
+      $tr = $('<tr>').appendTo(this.$table);
+      for (var col = 0; col < cols; col++) {
+        date = moment(weekStart).add(col,'weeks');
+        // Mark first day of week
+        cellID = user.attributes.id+'-'+date.format('YY-MM-DD');
+        $td = $('<td>').appendTo($tr)
+          .attr('id', cellID)
+          .addClass('left-cell');
+        // Three days
+        for (var i = 0; i < 3; i++) {
+          cellID = user.attributes.id+'-'+date.add(1,'day').format('YY-MM-DD');
+          $td = $('<td>').appendTo($tr)
+            .attr('id', cellID);
+        }
+        // Mark last day of week
+        cellID = user.attributes.id+'-'+date.add(1,'day').format('YY-MM-DD');
+        $td = $('<td>').appendTo($tr)
+          .attr('id', cellID)
+          .addClass('right-cell');
+      }
+    }, this);
+  },
+
+  markVacations: function(collection) {
+    _.each(collection, function(vacations) {
+      if (!_.isEmpty(vacations)) {
+        _.each(vacations, function(vacation) {
+          this.markVacation(vacation);
+        }, this);
+      }
+    }, this);
+  },
+
+  markVacation: function(vacation) {
+    var date = null,
+        endDate = null,
+        weekRange = null,
+        weekendCounter = 0,
+        vacationRange = null,
+        beginDate = moment(vacation.start),
+        duration  = vacation.duration,
+        userID    = vacation.user_id,
+        type      = vacation.kind;
+
+    for (date = beginDate.clone(); date < moment(beginDate).add(duration, 'days'); date.add(1, 'day')) {
+      if (App.Helpers.isWeekend(date.toDate())) {
+        duration++;
+        continue;
+      }
+      selector = '#'+userID+'-'+date.format('YY-MM-DD');
+      $(selector).addClass(vacation.kind);
+      $(selector).addClass(vacation.status);
+    }
+  },
+
+
+  // ***************************** Helpers *************************************
```

Comment:
Let me explain a bit :)

The `updateTimeTableDateRange()` method is a helper for the draft solution only. It is to be removed. The date range will be passed in by `TimeTables` view as options.

The `getNumberOfWeeks()` method is for the local use only as well as `calculateTableWidth()`.

There must be the `TimeTableByDay` view. I suppose it should have similar methods.

I plan refactoring after `Dashboard` is ready, and extract all the helpers in scope of this major refactoring.